socorro/external/postgresql/new_crash_source.py

```diff
@@ -0,0 +1,242 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import timedelta
+from collections import Sequence
+
+from configman import Namespace, RequiredConfig, class_converter
+
+from socorro.lib.converters import (
+    change_default,
+    two_level_nested_list_converter
+)
+from socorro.lib.datetimeutil import (
+    utc_now,
+    string_to_datetime,
+)
+from socorro.external.postgresql.dbapi2_util import execute_query_fetchall
+
+
+#==============================================================================
+class PGQueryNewCrashSource(RequiredConfig):
+    """This class is an iterator that will yield a stream of crash_ids based
+    on a query to the PG database."""
+    required_config = Namespace()
+    required_config.add_option(
+        'transaction_executor_class',
+        default="socorro.database.transaction_executor."
+        "TransactionExecutorWithInfiniteBackoff",
+        doc='a class that will manage transactions',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'database_class',
+        default='socorro.external.postgresql.connection_context'
+            '.ConnectionContext',
+        doc='the class responsible for connecting to Postgres',
+        from_string_converter=class_converter,
+        reference_value_from='resource.postgresql',
+    )
+    required_config.add_option(
+        'crash_id_query',
+        doc='sql to get a list of crash_ids',
+        default="select 'some_id'",
+        likely_to_be_changed=True,
+    )
+
+    #--------------------------------------------------------------------------
+    def __init__(self, config, name, quit_check_callback=None):
+
+        self.database = config.database_class(
+            config
+        )
+        self.transaction = config.transaction_executor_class(
+            config,
+            self.database,
+            quit_check_callback=quit_check_callback
+        )
+        self.config = config
+        self.name = name
+        self.data = ()
+        self.crash_id_query = config.crash_id_query
+
+    #--------------------------------------------------------------------------
+    def __iter__(self):
+        crash_ids = self.transaction(
+            execute_query_fetchall,
+            self.crash_id_query,
+            self.data
+        )
+
+        for a_crash_id in crash_ids:
+            yield a_crash_id
+
+    #--------------------------------------------------------------------------
+    def close(self):
+        self.database.close()
+
+    #--------------------------------------------------------------------------
+    new_crashes = __iter__
```

Comment:
This looks like a legacy/backwards compat fix where the name has evolved over time. 
Perhaps this is something of FTS that I don't get but can we stick to just one name?